apps/07jquery-tic-tac-toe/script.js

```diff
@@ -1,5 +1,63 @@
 'use strict';
 
 $(document).on('ready', function() {
-    // Put app logic in here
+
+  var playerTurn = 'X';
+
+  $('[data-cell]').click(function () {
+    $(this).text(playerTurn)
+    checkForWin();
+    playerTurn = (playerTurn === 'X') ? 'O' : 'X';
+  });
+
+  function horizontalWin() {
+    if ($('[data-cell = "0"]').text() === playerTurn &&
+        $('[data-cell = "1"]').text() === playerTurn &&
+        $('[data-cell = "2"]').text() === playerTurn ||
+        $('[data-cell = "3"]').text() === playerTurn &&
+        $('[data-cell = "4"]').text() === playerTurn &&
+        $('[data-cell = "5"]').text() === playerTurn ||
+        $('[data-cell = "6"]').text() === playerTurn &&
+        $('[data-cell = "7"]').text() === playerTurn &&
+        $('[data-cell = "8"]').text() === playerTurn) {
+          return true;
+        }
+  }
+
+  function verticalWin() {
+    if ($('[data-cell = "0"]').text() === playerTurn &&
+        $('[data-cell = "3"]').text() === playerTurn &&
+        $('[data-cell = "6"]').text() === playerTurn ||
+        $('[data-cell = "1"]').text() === playerTurn &&
+        $('[data-cell = "4"]').text() === playerTurn &&
+        $('[data-cell = "7"]').text() === playerTurn ||
+        $('[data-cell = "2"]').text() === playerTurn &&
+        $('[data-cell = "5"]').text() === playerTurn &&
+        $('[data-cell = "8"]').text() === playerTurn) {
+          return true;
+        }
+  }
+
+  function diagonalWin() {
+    if ($('[data-cell = "0"]').text() === playerTurn &&
+        $('[data-cell = "4"]').text() === playerTurn &&
+        $('[data-cell = "8"]').text() === playerTurn ||
+        $('[data-cell = "2"]').text() === playerTurn &&
+        $('[data-cell = "4"]').text() === playerTurn &&
+        $('[data-cell = "6"]').text() === playerTurn) {
+          return true;
+        }
+  }
+
+  function checkForWin() {
+      if (horizontalWin() === true || verticalWin() === true || diagonalWin() === true) {
```

Comment:
You could also write if(horizontalWin() || verticalWin() || diagonalWin() { ...}.

This will evaluate the truthiness of each function.